pkg/controller/vsphere/actuator.go

```diff
@@ -27,33 +28,36 @@ const (
 
 // Actuator is responsible for performing machine reconciliation.
 type Actuator struct {
-	client                     runtimeclient.Client
-	apiReader                  runtimeclient.Reader
-	eventRecorder              record.EventRecorder
-	TaskIDCache                map[string]string
-	StaticIPFeatureGateEnabled bool
-	openshiftConfigNamespace   string
+	client                             runtimeclient.Client
+	apiReader                          runtimeclient.Reader
+	eventRecorder                      record.EventRecorder
+	TaskIDCache                        map[string]string
+	StaticIPFeatureGateEnabled         bool
+	HostVMGroupZonalFeatureGateEnabled bool
+	openshiftConfigNamespace           string
 }
 
 // ActuatorParams holds parameter information for Actuator.
 type ActuatorParams struct {
-	Client                     runtimeclient.Client
-	APIReader                  runtimeclient.Reader
-	EventRecorder              record.EventRecorder
-	TaskIDCache                map[string]string
-	StaticIPFeatureGateEnabled bool
-	OpenshiftConfigNamespace   string
+	Client                             runtimeclient.Client
+	APIReader                          runtimeclient.Reader
+	EventRecorder                      record.EventRecorder
+	TaskIDCache                        map[string]string
+	StaticIPFeatureGateEnabled         bool
```

Comment:
Instead of updating all the parms over time for each new feature gate, would it make sense to pass the list of enabled feature gates and let the section of code pull out what it needs? I'm doing this over here in my multi disk PR: https://github.com/openshift/machine-api-operator/pull/1290/files

Reply:
@JoelSpeed I think what @vr4manta mentions here with feature gates makes sense but can I get your buy in before I go and refactor ;-)

Reply:
I agree with Neil, typically we would see `featuregates.Enabled(configv1.MyFeatureGate)` at the call site. 

I believe since this is in the vSphere actuator, this is constructed from a list of arguments passed to the cmd, so there should be prior art here

I believe @theobarberbany did some work on this recently